o1vm/src/pickles/prover.rs

```diff
@@ -0,0 +1,221 @@
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Round 2: Creating and committing to the quotient polynomial
+    ////////////////////////////////////////////////////////////////////////////
+
+    // FIXME: add quotient polynomial
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Round 3: Evaluations at ζ and ζω
+    ////////////////////////////////////////////////////////////////////////////
+
+    let zeta_chal = ScalarChallenge(fq_sponge.challenge());
+    let (_, endo_r) = G::endos();
+
+    let zeta = zeta_chal.to_field(endo_r);
+    let zeta_omega = zeta * omega;
+
+    // FIXME: add selectors
+    let evals = |point| {
+        let WitnessColumns {
+            scratch,
+            instruction_counter,
+            error,
+            // FIXME
+            selector: _,
+        } = &polys;
+        let eval = |poly: &DensePolynomial<G::ScalarField>| poly.evaluate(point);
+        let scratch = scratch.par_iter().map(eval).collect::<Vec<_>>();
+        WitnessColumns {
+            scratch: scratch.try_into().unwrap(),
+            instruction_counter: eval(instruction_counter),
+            error: eval(error),
+            // FIXME
+            selector: eval(error),
+        }
+    };
+    // All evaluations at ζ
+    let zeta_evaluations = evals(&zeta);
+    // All evaluations at ζω
+    let zeta_omega_evaluations = evals(&zeta_omega);
+
+    // Absorbing evaluations with a sponge for the other field
+    // We initialize the state with the previous state of the fq_sponge
+    let fq_sponge_before_evaluations = fq_sponge.clone();
+    let mut fr_sponge = EFrSponge::new(G::sponge_params());
+    fr_sponge.absorb(&fq_sponge.digest());
+
+    for (zeta_eval, zeta_omega_eval) in zeta_evaluations
+        .scratch
+        .iter()
+        .zip(zeta_omega_evaluations.scratch.iter())
+    {
+        fr_sponge.absorb(zeta_eval);
+        fr_sponge.absorb(zeta_omega_eval);
+    }
+    fr_sponge.absorb(&zeta_evaluations.instruction_counter);
+    fr_sponge.absorb(&zeta_omega_evaluations.instruction_counter);
+    fr_sponge.absorb(&zeta_evaluations.error);
+    fr_sponge.absorb(&zeta_omega_evaluations.error);
+    // FIXME: add selectors
+
+    ////////////////////////////////////////////////////////////////////////////
+    // Round 4: Opening proof w/o linearization polynomial
+    ////////////////////////////////////////////////////////////////////////////
+
+    // Preparing the polynomials for the opening proof
+    let mut polynomials: Vec<_> = polys.scratch.into_iter().collect();
+    polynomials.push(polys.instruction_counter);
+    polynomials.push(polys.error);
+    // FIXME: add selectors
+    let polynomials: Vec<_> = polynomials
+        .iter()
+        .map(|poly| {
+            (
+                DensePolynomialOrEvaluations::DensePolynomial(poly),
+                // We do not have any blinder, therefore we set to 0.
```

Comment:
I thought the field PolyComm contains the commitment
This comment makes me think it contains the blinder

Reply:
As discussed earlier this week, PolyComm is abstracted over any type `T`, and sometimes a scalar is used.